skypy/galaxy/redshift.py

```diff
@@ -5,243 +5,216 @@
 
-    References
-    ----------
-    .. [1] Smail I., Ellis R. S., Fitchett M. J., 1994, MNRAS, 270, 245
-    .. [2] Amara A., Refregier A., 2007, MNRAS, 381, 1018
+@uses_default_cosmology
+@dependent_argument('M_star', 'redshift')
+@dependent_argument('phi_star', 'redshift')
+@dependent_argument('alpha', 'redshift')
+@broadcast_arguments('redshift', 'M_star', 'phi_star', 'alpha')
+def schechter_lf_redshift(redshift, M_star, phi_star, alpha, m_lim, fsky, cosmology, noise=True):
+    r'''Sample redshifts from Schechter luminosity function.
 
-    Examples
-    --------
-    Sample 10 random variates from the Smail model with `alpha = 1.5` and
-    `beta = 2` and median redshift `z_median = 1.2`.
-
-    >>> from skypy.galaxy.redshift import smail
-    >>> redshift = smail(1.2, 1.5, 2.0, size=10)
-
-    '''
-
-    k = (alpha+1)/beta
-    t = z_median**beta/sc.gammainccinv(k, 0.5)
-    g = np.random.gamma(shape=k, scale=t, size=size)
-    return g**(1/beta)
-
-
-def herbel_redshift(alpha, a_phi, b_phi, a_m, b_m, cosmology, low=0.0,
-                    high=2.0,
-                    size=None, absolute_magnitude_max=-16.0, resolution=100):
-    r""" Redshift following the Schechter luminosity function marginalised over
-    luminosities following the Herbel et al. [1]_ model.
+    Sample the redshifts of galaxies following a Schechter luminosity function
+    with potentially redshift-dependent parameters, limited by an apparent
+    magnitude `m_lim`, for a fraction `fsky` of the sky.
 
     Parameters
     ----------
-    alpha : float or scalar
-        The alpha parameter in the Schechter luminosity function
-    a_phi, b_phi : float or scalar
-        Parametrisation factors of the normalisation factor Phi_* as a function
-        of redshift according to Herbel et al. [1]_ equation (3.4).
-    a_m, b_m : float or scalar
-        Parametrisation factors of the characteristic absolute magnitude M_* as
-        a function of redshift according to Herbel et al. [1]_ equation (3.3).
-    cosmology : instance
-        Instance of an Astropy Cosmology class.
-    low : float or array_like of floats, optional
-        The lower boundary of teh output interval. All values generated will be
-        greater than or equal to low.
-        The default value is 0.0.
-    high : float or array_like of floats, optional
-        Upper boundary of output interval. All values generated will be less
-        than high. The default value is 2.0
-    size : int or tuple of ints, optional
-        The number of redshifts to sample. If None one value is sampled.
-    absolute_magnitude_max : float or scalar, optional
-        Upper limit of the considered absolute magnitudes of the galaxies to
-        wanna sample.
-    resolution : int, optional
-        Characterises the resolution of the sampling. Default is 100.
+    redshift : array_like
+        Input redshift range at which the Schechter function parameters are
```

Comment:
```suggestion
        Input redshift grid on which the Schechter function parameters are
```
This confused me for ages as I thought it was expecting a literal range i.e. `redshift=[z_min, z_max]`. Hopefully this makes it clearer